vignettes/cancensus.Rmd

```diff
@@ -0,0 +1,195 @@
+# Returns a data frame with data only
+census_data <- get_census(dataset='CA16', regions=list(CMA="59933"),
+                          vectors=c("v_CA16_408","v_CA16_409","v_CA16_410"),
+                          level='CSD', use_cache = FALSE, geo_format = NA)
+
+# Returns data and geography as an sf-class data frame
+census_data <- get_census(dataset='CA16', regions=list(CMA="59933"),
+                          vectors=c("v_CA16_408","v_CA16_409","v_CA16_410"),
+                          level='CSD', use_cache = FALSE, geo_format = 'sf')
+
+# Returns a SpatialPolygonsDataFrame object with data and geogrpahy
+census_data <- get_census(dataset='CA16', regions=list(CMA="59933"),
+                          vectors=c("v_CA16_408","v_CA16_409","v_CA16_410"),
+                          level='CSD', use_cache = FALSE, geo_format = 'sp')
+```
+
+**cancensus** utilizes caching to increase speed, minimize API token usage, and to make data available offline. Downloaded data is hashed and stored locally so if a call is made to access the same data, **cancensus** will read the local version instead. To force **cancensus** to refresh the data, specify `use_cache = FALSE` as a parameter for `get_census`. 
+
+Additional parameters for advanced options can be viewed by running `?get_census`.
+
+## Census Datasets
+
+**cancensus** can access Statistics Canada Census data for the 2006 Census, the 2011 Census and National Household Survey, as well as the latest available data from the 2016 Census. Additional data for the 2016 Census will be included in Censusmapper within a day or two after public release by Statistics Canada. Statistics Canada maintains a release schedule for the Census 2016 Program which can be viewed on their [website](http://www12.statcan.gc.ca/census-recensement/2016/ref/release-dates-diffusion-eng.cfm). 
+
+To view available datasets, run `list_census_datasets()`.
+
+```{r list datasets, message=FALSE, warning=FALSE}
+list_census_datasets()
+```
+
+As other Census datasets become available via the CensusMapper API, they will be listed as output when calling `list_census_datasets()`. 
+
+## Census Regions
+
+Census data is aggregated at multiple [geographic levels](### Census Geographic Levels). Census geographies at the national (C), provincial (PR), census metropolitan area (CMA), census division (CD), and census subdivision (CSD) are defined as named census regions. 
+
+Canadian Census geography can change in between Census periods. **cancensus** provides a function, `list_census_regions(dataset)`, to display all named census regions and their corresponding id for a given census dataset. 
+
+```{r list regions, message=FALSE, warning=FALSE, eval = FALSE}
+list_census_regions("CA16")
+```
+
+The `regions` parameter in `get_census` requires as input a list of region id strings that correspond to that regions geoid. You can combine different regions together into region lists.
+
+```{r, message=FALSE, warning=FALSE, eval=FALSE}
```

Comment:
Again why is this not evaluated?

Reply:
Same as above, produced a ton of messaging. Once #98 is implemented, there's less noise and we can evaluate these if we want.

Reply:
This has been rebase onto master, so it includes the code from #98 now -- check it out locally. If anything, spewing messages even if the user is passing `quietly = TRUE` should be a bug. I'd support filing unwanted output as a separate issue.